code/_onclick/hud/spell_screen_objects.dm

```diff
@@ -144,7 +144,7 @@
 	..()
 	if(spellmaster)
 		spellmaster.spell_objects -= src
-	if(spellmaster && !spellmaster.spell_objects.len)
+	if(spellmaster && !spellmaster.spell_objects.len && master)
```

Comment:
master is a thing that /obj/screen has, and from what I'm now remembering, it's used to refer to things that hud elements actually reference (i.e. an icon on your hud for a welding tool points to an actual welding tool contained by you somewhere) or something along those lines.

I don't know what it'd be for spells, or if it's even set. Maybe it should be returnToPool(spellmaster)